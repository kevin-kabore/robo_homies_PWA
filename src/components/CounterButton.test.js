import { shallow } from 'enzyme';
import React from 'react';

import CounterButton from './CounterButton';

describe('CounterButton Component', () => {
	it('Will render the Component with state', () => {
		// snapshot testing used for all presentational/stateless components the whole component
		const mockColor = 'red';
		const wrapper = shallow(<CounterButton color={mockColor}/>);
		
		expect(wrapper).toMatchSnapshot()
		expect(wrapper.state('count')).toEqual(0);
	})

	it('Will correctly increment the count', () => {
		const mockColor = 'red';
		const wrapper = shallow(<CounterButton color={mockColor}/>);
		
		wrapper.simulate('click');
		wrapper.simulate('click');
		expect(wrapper.state('count')).toEqual(2);
		// expect(wrapper.state('count')).toEqual(1);
	})
})
