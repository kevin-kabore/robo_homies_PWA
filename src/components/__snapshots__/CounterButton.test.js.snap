// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton Component Will render the Component with state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CounterButton
    color="red"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Count: ",
        0,
      ],
      "color": "red",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "Count: ",
      0,
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Count: ",
          0,
        ],
        "color": "red",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Count: ",
        0,
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
